test/e2e/framework/provider_less.go

```diff
@@ -0,0 +1,27 @@
+//go:build providerless
+// +build providerless
+
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+func init() {
+	// fake "gce"
+	RegisterProvider("gce", func() (ProviderInterface, error) {
```

Comment:
this sounds very hacky , can we try to make it work without this hack? what tests are failing without this?

Reply:
Anything with `--provider=gce` will fail - https://cs.k8s.io/?q=%5C-%5C-provider%3Dgce&i=nope&files=&excludeFiles=&repos=kubernetes/test-infra

there are a bunch of presubmits that use this

Reply:
it feels like is something that can bit us later, maybe we can rework the tests instead?

Reply:
@aojea yep just don't want to do in this PR.

Reply:
We have a similar issue with nothing vs "skeleton", it's going to take a LOT more work to disentangle this provider concept from the tests. This makes sense as a stop-gap IMHO.